extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -102,6 +104,33 @@ AdditionalBeanBuildItem beans() {
                 QuarkusWorkerPoolRegistry.class);
     }
 
+    @BuildStep
+    void defaultConnectorSettings(List<ConnectorProviderBuildItem> connectorProviderBuildItems,
+            CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> defaultBuildItemBuildProducer) {
+        if (connectorProviderBuildItems.size() != 1) {
+            //more than one connector
+            return;
+        }
+        var config = ConfigProvider.getConfig();
+        for (var an : indexBuildItem.getIndex().getAnnotations(ReactiveMessagingDotNames.INCOMING)) {
+            String name = an.value().asString();
+            String prop = "mp.messaging.incoming." + name + ".connector";
+            if (config.getOptionalValue(prop, String.class).isEmpty()) {
+                defaultBuildItemBuildProducer
+                        .produce(new RunTimeConfigurationDefaultBuildItem(prop, connectorProviderBuildItems.get(0).getName()));
+            }
+        }
+        for (var an : indexBuildItem.getIndex().getAnnotations(ReactiveMessagingDotNames.OUTGOING)) {
+            String name = an.value().asString();
+            String prop = "mp.messaging.outgoing." + name + ".connector";
+            if (config.getOptionalValue(prop, String.class).isEmpty()) {
+                defaultBuildItemBuildProducer
+                        .produce(new RunTimeConfigurationDefaultBuildItem(prop, connectorProviderBuildItems.get(0).getName()));
+            }
+        }
```

Comment:
This is not enough, you also need to look at the `@Channel` injection points. If the injection point is of type `Emitter` or `MutinyEmitter`, it's an "outgoing", if it is a `Multi` (or `Publisher` or `PublisherBuilder` :-) ), it's an "incoming".

(Or something like that. I always have to look up the specifics.)

Reply:
So, you need to look for "un-handled" channels, which is even more tricky. 
On top of my head, only the wiring API can provide you such data. 

For example, it's not because you have an emitter that you must have the configuration as the emitter may be consumed internally. 

I'm wondering if we shouldn't try to implement this in the wiring API of reactive messaging directly as it has the graph and so can detect "loose configuration"

Reply:
Ah ah, good point!

Reply:
Does @Channel work for stuff other than Kafka? I have support for it in Kafka but this was in the Kafka processor, the support does not seem to be generic.